fuel-tx/src/builder.rs

```diff
@@ -276,25 +276,25 @@ impl<Tx: Buildable> TransactionBuilder<Tx> {
     }
 
     #[cfg(feature = "std")]
-    pub fn _finalize(&mut self) -> Tx {
+    pub fn _finalize(&mut self, parameters: &ConsensusParameters) -> Tx {
```

Comment:
It is better if the builder stores the `ConsensusParameters` inside. In this case, we don't need to propagate the parameter everywhere and break the API. 

We can use `Builder.set_consensus_parametrs` in tests where we want to use it. The same pattern that we use in the `fuel-core` where we have `Option<ConsensusParameters>` field=)